### PR TITLE
update for vxWorks

### PR DESCRIPTION
**Pull request by BaoshanPang**
1. support crt-static 
2. change armv7_wrs_vxworks to armv7_wrs_vxworks_eabihf. 
3. code cleanup

### CODE REVIEW

File: src/librustc_target/spec/powerpc_wrs_vxworks_spe.rs [hunk 1]
```diff
@@ -20,7 +20,6 @@ pub fn target() -> TargetResult {
         options: TargetOptions {
             // feature msync would disable instruction 'fsync' which is not supported by fsl_p1p2
             features: "+secure-plt,+msync".to_string(),
-            target_mcount: "_mcount".to_string(),
```
Comment (n-salim) [inline src/librustc_target/spec/powerpc_wrs_vxworks_spe.rs]: "mcount" (or the name we specify here) is the function called by compiler generated instrumentation code when profiling is enabled. 

https://www.kernel.org/doc/Documentation/trace/ftrace-design.txt
https://ftp.gnu.org/pub/old-gnu/Manuals/gprof-2.9.1/html_node/gprof_25.html

VxWorks would need to implement this function (we don't currently have it). Removing this code doesn't actually disable calls to mcount, it just reverts to the default name "mcount". 

For clarity I suggest adding a line to vxworks_base.rs:

`target_mcount: "_mcount".to_string(), // VxWorks needs to implement this to support profiling`

File: src/librustc_target/spec/powerpc_wrs_vxworks_spe.rs [hunk 2]
```diff
@@ -20,7 +20,6 @@ pub fn target() -> TargetResult {
         options: TargetOptions {
```
Comment (n-salim) [inline src/librustc_target/spec/powerpc_wrs_vxworks_spe.rs]: Move the following common settings from the target specific files to vxworks_base.rs:

- target_os
- target_vendor
- linker_flavor

File: src/librustc_target/spec/powerpc_wrs_vxworks_spe.rs (hunk 1 above)
Comment (BaoshanPang) [reply src/librustc_target/spec/powerpc_wrs_vxworks_spe.rs]: Done.

File: src/librustc_target/spec/powerpc_wrs_vxworks_spe.rs (hunk 2 above)
Comment (BaoshanPang) [reply src/librustc_target/spec/powerpc_wrs_vxworks_spe.rs]: Done. I also moved target_env to vxworks_base as they are all "gnu".

Comment (BaoshanPang) [reply src/librustc_target/spec/powerpc_wrs_vxworks_spe.rs]: Actually I need to revert the change here. The vxworks_base.rs suppose to hold all "options", but 
they doesn't belong to "options".

**Comment by BaoshanPang** [comment]
cc @bpangWR

File: src/tools/compiletest/src/runtest.rs [hunk 3]
```diff
@@ -1659,10 +1659,10 @@ impl<'test> TestCx<'test> {
             _ if self.config.target.contains("vxworks") => {
                 let aux_dir = self.aux_output_dir_name();
                 let ProcArgs { prog, args } = self.make_run_args();
-                let mut vx_run = Command::new("vx-run");
-                vx_run.args(&[&prog]).args(args).envs(env.clone());
+                let mut wr_run = Command::new("wr-run");
+                wr_run.args(&[&prog]).args(args).envs(env.clone());
                 self.compose_and_run(
-                    vx_run,
+                    wr_run,
```
Comment (n-salim) [inline src/tools/compiletest/src/runtest.rs]: What is vx-run (or wr-run)?

File: src/librustc_target/spec/powerpc_wrs_vxworks_spe.rs (hunk 2 above)
Comment (n-salim) [reply src/librustc_target/spec/powerpc_wrs_vxworks_spe.rs]: > Actually I need to revert the change here
OK that's unfortunate.

File: src/tools/compiletest/src/runtest.rs (hunk 3 above)
Comment (BaoshanPang) [reply src/tools/compiletest/src/runtest.rs]: This is the script I used to run test case on board(Simics, Vxsim or real board).

**Pull request merged**